src/components/TokenGeneralInfo.js

```diff
@@ -0,0 +1,182 @@
+      return `${configArr[0]}:${configArr[1]}...${configArr[3]}`;
+    }
+
+    const renderTokenInfo = () => {
+      return (
+        <div className="token-general-info">
+          <p className="mb-2"><strong>UID: </strong>{this.props.token.uid}</p>
+          <p className="mt-2 mb-2"><strong>Name: </strong>{this.props.token.name}</p>
+          <p className="mt-2 mb-2"><strong>Symbol: </strong>{this.props.token.symbol}</p>
+          <p className="mt-2 mb-2"><strong>Total supply: </strong>{hathorLib.helpers.prettyValue(this.state.totalSupply)} {this.props.token.symbol}</p>
+          <p className="mt-2 mb-0"><strong>Can mint new tokens: </strong>{this.state.canMint ? 'Yes' : 'No'}</p>
+          <p className="mb-2 subtitle">Indicates whether the token owner can create new tokens, increasing the total supply</p>
+          <p className="mt-2 mb-0"><strong>Can melt tokens: </strong>{this.state.canMelt ? 'Yes' : 'No'}</p>
+          <p className="mb-2 subtitle">Indicates whether the token owner can destroy tokens, decreasing the total supply</p>
+          <p className="mt-2 mb-4"><strong>Total number of transactions: </strong>{this.state.transactionsCount}</p>
+        </div>
+      );
+    }
+
+    const renderConfigString = () => {
+      return (
+        <div className='d-flex flex-row align-items-center justify-content-center mt-4 w-100'>
+          <div className='d-flex flex-column align-items-center config-string-wrapper'>
+            <p><strong>Configuration String</strong></p>
+            <span ref="configurationString" className="mb-2">
+              {getShortConfigurationString()}
+              <CopyToClipboard text={configurationString} onCopy={this.copied}>
+                <i className="fa fa-clone pointer ml-1" title="Copy to clipboard"></i>
+              </CopyToClipboard>
+            </span> 
+            <QRCode size={200} value={configurationString} />
+            <a className="mt-2" onClick={(e) => this.downloadQrCode(e)} download={`${this.props.token.name} (${this.props.token.symbol}) - ${configurationString}`} href="true" ref="downloadLink">Download <i className="fa fa-download ml-1" title="Download QRCode"></i></a>
+          </div>
+        </div>
+      );
+    }
+
+    return (
+      <div className="flex align-items-center">
+        <div className='d-flex flex-column align-items-start justify-content-between token-detail-top'>
+          <div className='d-flex flex-column justify-content-between mr-3'>
+            {renderTokenInfo()}
+          </div>
+          {this.props.showConfigString && renderConfigString()}
+        </div>
+        <HathorAlert ref="alertSuccess" text={this.state.successMessage} type="success" />
+      </div>
+    )
+  }
+}
+
+/*
+ * token: Token to show the information {name, symbol, uid}
+ * showConfigString: If should show the configuration string of the token with the qrcode
+ */
+TokenGeneralInfo.propTypes = {
+  token: PropTypes.object.isRequired,
```

Comment:
Done

Reply:
@msbrogli I need one check here. You've suggested to use the exact object on this prop types, however I use this component in two different places.

1. About token: it's perfect using the exact;
2. TransactionDetail (when opening the unregistered token modal): in this page I already use a token object that has {'name', 'uid', 'symbol', 'unknown'} and reuse it to send to this component (`TokenGeneralInfo`). However this is not the exact object we are expecting (we have the 'unknown' as a plus and it throws a warning.

I thought about two solutions:

1. Send to the component an updated version of the object, where I remove the 'unknown' key;
2. Update the prop types, to use shape instead of exact (https://reactjs.org/docs/typechecking-with-proptypes.html), so I can have a 'superobject' also in this props, with all needed keys and some optional more.

I feel the best option is the first, so we define what we want and all components must adjust but I don't know if the code will look like a workaround removing a key to send an object as props.

What do you think?

Reply:
I think it is ok to use the `shape` instead of `exact`. I don't see any problem to get "superobjects", and I think it is simpler and less error prone than creating "partial objects".

Reply:
I've updated to use `shape`.

We have now:

```
TokenGeneralInfo.propTypes = { 
  token: PropTypes.shape({
    name: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
    uid: PropTypes.string.isRequired,
  }), 
  showConfigString: PropTypes.bool.isRequired,
}
```